youtube_dl/extractor/chaturbate.py

```diff
@@ -32,7 +32,7 @@ def _real_extract(self, url):
         webpage = self._download_webpage(url, video_id)
 
         m3u8_url = self._search_regex(
-            r'src=(["\'])(?P<url>http.+?\.m3u8.*?)\1', webpage,
+            r'var hlsSourceFast = (["\'])(?P<url>http.+?\.m3u8)', webpage,
```

Comment:
I did not ask to remove **all formats extraction** but to remove **code duplication**.